gc.md

```diff
@@ -1,65 +1,78 @@
+# V8 Garbage Collector
+
+_find the previous version of this document at
+[crankshaft/gc.md](crankshaft/gc.md)_
+
 <!-- START doctoc generated TOC please keep comment here to allow auto update -->
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
-**Table of Contents**  *generated with [DocToc](http://doctoc.herokuapp.com/)*
-
-- [v8 Garbage Collector](#v8-garbage-collector)
-  - [Goals, Techniques](#goals-techniques)
-  - [Cost of Allocating Memory](#cost-of-allocating-memory)
-  - [How objects are determined to be dead](#how-objects-are-determined-to-be-dead)
-  - [Two Generations](#two-generations)
-  - [Heap Organization in Detail](#heap-organization-in-detail)
-    - [New Space](#new-space)
-    - [Old Pointer Space](#old-pointer-space)
-    - [Old Data Space](#old-data-space)
-    - [Large Object Space](#large-object-space)
-    - [Code Space](#code-space)
-    - [Cell Space, Property Cell Space, Map Space](#cell-space-property-cell-space-map-space)
-    - [Pages](#pages)
-  - [Young Generation](#young-generation)
-    - [ToSpace, FromSpace, Memory Exhaustion](#tospace-fromspace-memory-exhaustion)
-      - [Sample Scavenge Scenario](#sample-scavenge-scenario)
-        - [Collection to free ToSpace](#collection-to-free-tospace)
-      - [Write Barriers](#write-barriers)
-        - [Write Barrier Crankshaft Optimizations](#write-barrier-crankshaft-optimizations)
-      - [Considerations](#considerations)
-  - [Old Generation](#old-generation)
-    - [Collection Steps](#collection-steps)
-    - [Mark Sweep and Mark Compact](#mark-sweep-and-mark-compact)
-      - [Mark](#mark)
-      - [Marking State](#marking-state)
-        - [Depth-First Search](#depth-first-search)
-      - [Handling Deque Overflow](#handling-deque-overflow)
-      - [Sweep and Compact](#sweep-and-compact)
-      - [Sweep](#sweep)
-      - [Compact](#compact)
-    - [Incremental Mark and Lazy Sweep](#incremental-mark-and-lazy-sweep)
-      - [Incremental Marking](#incremental-marking)
-      - [Lazy Sweeping](#lazy-sweeping)
-  - [Causes For GC Pause](#causes-for-gc-pause)
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Goals, Techniques](#goals-techniques)
+- [Cost of Allocating Memory](#cost-of-allocating-memory)
+- [How objects are determined to be dead](#how-objects-are-determined-to-be-dead)
+- [Two Generations](#two-generations)
+  - [Generational Garbage Collector](#generational-garbage-collector)
+- [Heap Organization in Detail](#heap-organization-in-detail)
+  - [New Space aka Young Generation](#new-space-aka-young-generation)
+  - [Old Pointer Space](#old-pointer-space)
+  - [Old Data Space](#old-data-space)
+  - [Large Object Space](#large-object-space)
+  - [Code Space](#code-space)
+  - [Cell Space, Property Cell Space, Map Space](#cell-space-property-cell-space-map-space)
```

Comment:
Cell and Property spaces were merged to Old space.

Reply:
Addressed in 9d79c90